CodeGeneration/Sources/SyntaxSupport/GrammarGenerator.swift

```diff
@@ -60,4 +60,20 @@ struct GrammarGenerator {
       .map { " - `\($0.varOrCaseName)`: \(generator.grammar(for: $0))" }
       .joined(separator: "\n")
   }
+
+  /// Generates the list of possible token kinds for this particular ``Child``.
+  /// The child must be of kind ``ChildKind/token``. Otherwise, an empty string will be returned.
+  static func childTokenChoices(for child: Child) -> String {
+    let generator = GrammarGenerator()
+
+    if case .token(let choices, _, _) = child.kind {
+      if choices.count == 1 {
+        return " \(generator.grammar(for: choices.first!))"
+      } else {
+        return choices.map { " - \(generator.grammar(for: $0))" }.joined(separator: "\n")
+      }
+    } else {
+      return ""
```

Comment:
This is sus: if the `child` is not a `.token`, then its token choices are not an empty string, they are technically `nil`. No practical harm out of that, perhaps that's just a technicality. Also mentioned in the docc comment of the function.

@ahoppen, want me to make this func return an optional string, and unwrap upstream?

Reply:
I actually even started with `precondition` at first, and then thought that if I'm carefully unwrapping, it's not necesary. 

But, 💯, it's better to have an explicit failure than an unexpected empty output.